pymc3/distributions/continuous.py

```diff
@@ -1653,6 +1654,58 @@ def logcdf(self, value):
             tt.switch(tt.gt(y, 1), tt.log1p(-0.5 * tt.exp(-y)), tt.log(1 - 0.5 * tt.exp(-y))),
         )
 
+class AsymmetricLaplace(Continuous):
+    """
+    Assymetric Laplace Distribution
+    See also: https://en.wikipedia.org/wiki/Asymmetric_Laplace_distribution
+    """
+
+    def __init__(self, scale, symmetry, testval=0.0, *args, **kwargs):
+        """
+        Constructor
+        :param scale: scale parameter
+        :param symmetry: asymmetry parameter. Reduces to a normal laplace distribution with value 1
+        """
+        self.scale = tt.as_tensor_variable(scale)
+        self.symmetry = tt.as_tensor_variable(symmetry)
+
+        super().__init__(*args, **kwargs, testval=testval)
+
+    def random(self, point=None, size=None):
+        """
+        Draw random samples from this distribution, using the inverse CDF method.
+        :param point: not used
+        :param size: size of sample to draw
+        :return: Samples
+        """
+        if point is not None:
+            raise NotImplementedError('Random not implemented with point specified')
+
+        if size is not None:
+            u = np.random.uniform(size=size)
+            x = - tt.log((1 - u) * (1 + self.symmetry ** 2)) / (self.symmetry * self.scale) * (
+                    u > ((self.symmetry ** 2) / (1 + self.symmetry ** 2))) + self.symmetry * tt.log(
+                u * (1 + self.symmetry ** 2) / (self.symmetry ** 2)) / self.scale * (
+                        u < ((self.symmetry ** 2) / (1 + self.symmetry ** 2)))
+
+            return x
+
+        u = np.random.uniform()
+        if u > (self.symmetry ** 2) / (1 + self.symmetry ** 2):
+            x = - tt.log((1 - u) * (1 + self.symmetry ** 2)) / (self.symmetry * self.scale)
```

Comment:
After we have concrete values as numpy arrays, its better to use numpy equivalents for these tensor operations. Most of the random methods in PyMC3 follow this.
Otherwise, you need a `.eval()` at the end of random method to draw values.